src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/utils.tsx

```diff
@@ -0,0 +1,480 @@
+import { generatePath } from 'react-router-dom'
+import { GroupBase, OptionsOrGroups } from 'react-select'
+
+import {
+    DEPLOYMENT_HISTORY_CONFIGURATION_LIST_MAP,
+    SelectPickerOptionType,
+    stringComparatorBySortOrder,
+    YAMLStringify,
+    DeploymentConfigDiffProps,
+    SortingOrder,
+    yamlComparatorBySortOrder,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { ReactComponent as ICCheck } from '@Icons/ic-check.svg'
+import { ReactComponent as ICStamp } from '@Icons/ic-stamp.svg'
+import { ReactComponent as ICEditFile } from '@Icons/ic-edit-file.svg'
+import { prepareHistoryData } from '@Components/app/details/cdDetails/service'
+import { TemplateListDTO, TemplateListType } from '@Components/deploymentConfig/types'
+import {
+    AppEnvDeploymentConfigDTO,
+    AppEnvDeploymentConfigQueryParamsType,
+    AppEnvDeploymentConfigType,
+    ConfigMapSecretDataConfigDatumDTO,
+    ConfigResourceType,
+    DeploymentTemplateDTO,
+} from '@Pages/Applications/DevtronApps/service.types'
+import { DraftState } from '@Pages/Shared/ConfigMapSecret/ConfigMapSecret.types'
+
+import {
+    EnvironmentOptionType,
+    EnvResourceType,
+    AppEnvDeploymentConfigQueryParams,
+    DeploymentConfigParams,
+} from '../../AppConfig.types'
+
+/**
+ * Retrieves the draft data from the given configuration data object.
+ *
+ * @param configData - The configuration data object.
+ * @returns The draft data if available, otherwise the original data.
+ */
+export const getDraftData = (configData: ConfigMapSecretDataConfigDatumDTO): ConfigMapSecretDataConfigDatumDTO => {
+    if (configData?.draftMetadata) {
+        const parsedData = JSON.parse(configData.draftMetadata.data)
+        return { ...parsedData.configData[0], draftMetadata: configData.draftMetadata }
```

Comment:
check if null check is requiered here for configData?.[0]

Reply:
no check required as this will be have data 
null check already covered by line 43